java/pom.xml

```diff
@@ -144,11 +144,23 @@
         <skipNativeCopy>false</skipNativeCopy>
         <cxx11.abi.value>ON</cxx11.abi.value>
         <cxx.flags/>
+        <CMAKE_EXPORT_COMPILE_COMMANDS>OFF</CMAKE_EXPORT_COMPILE_COMMANDS>
         <CUDA_STATIC_RUNTIME>OFF</CUDA_STATIC_RUNTIME>
         <PER_THREAD_DEFAULT_STREAM>OFF</PER_THREAD_DEFAULT_STREAM>
+        <project.build.directory>target</project.build.directory>
+        <native.build.path>${project.build.directory}/cmake-build</native.build.path>
     </properties>
 
     <profiles>
+        <profile> 
+            <id>managed-release-directory</id>
+            <activation> 
+                <activeByDefault>false</activeByDefault>
+            </activation>
+            <properties>
+                <project.build.directory>target/cuda-${CUDA_VERSION}/${git.branch}/release</project.build.directory>
```

Comment:
Where did `CUDA_VERSION` and `git.branch` come from?

Reply:
They will be set in the docker for rapids-compose.

Reply:
Is rapids-compose now a requirement to build this?

Reply:
No, here's what I am trying to achieve. when you set the `CUDA_VERSION` and `git.branch` variables and call this profile. It will put the artifacts in `target/cuda-${CUDA_VERSION}/${git.branch}/release`. 
Ideally I didn't want to change the pom at all but the `project.build.output` can't be set from the command line. 

I think it will be better if I focus this PR only on the rapids-compose change and undo the change to the output directory.

Reply:
It shouldn't be, and imo we should avoid adding references to values exclusive to the rapids-compose environment.

The intention is to prefix the C++ build directories with `cuda_version/branch_name/build_type`, because it helps improve (or avoid) costly rebuild times changing those values. We can achieve this in compose by setting `-Dnative.build.path` in the `mvn package` command.

Since Java rebuild times are fast compared to C++, I think it's overkill to prefix the entire `target` directory of Java build artifacts with those values.